google/datalab/bigquery/commands/_bigquery.py

```diff
@@ -250,23 +250,31 @@ def _create_load_subparser(parser):
   return load_parser
 
 
-def _default_context_for_args(args):
-  """Get the default Context for the parsed arguments.
+def _construct_context_for_args(args):
+  """Construct a new Context for the parsed arguments.
 
   Args:
     args: the dictionary of magic arguments.
   Returns:
-    A Context based on the current default context, but with any explicitly specified
-      arguments overriding the defaults.
+    A new Context based on the current default context, but with any explicitly
+      specified arguments overriding the default's config.
   """
-  context = google.datalab.Context.default()
+  global_default_context = google.datalab.Context.default()
+  config = {}
+  for key in global_default_context.config:
+    config[key] = global_default_context.config[key]
+
   dialect_arg = args.get('dialect', None)
   billing_tier_arg = args.get('billing', None)
   if dialect_arg:
-    context.config['bigquery_dialect'] = dialect_arg
+    config['bigquery_dialect'] = dialect_arg
```

Comment:
why pick these two config values individually? we might add things in the future, can't we just copy all the keys in `args` over the ones in `config`?